sdk/communication/Azure.Communication.Calling.Server/src/ContentDownloader.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Communication.Calling.Server.Models;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Calling.Server
+{
+    internal class ContentDownloader
+    {
+        private readonly ConversationClient _client;
+
+        internal ContentDownloader(ConversationClient client)
+        {
+            _client = client;
+        }
+
+        internal async Task<Response<Stream>> DownloadStreamingInternal(Uri endpoint, HttpRange range, bool async, CancellationToken cancellationToken)
+        {
+            using DiagnosticScope scope = _client._clientDiagnostics.CreateScope($"{nameof(ConversationClient)}.{nameof(DownloadStreamingInternal)}");
```

Comment:
```suggestion
            using DiagnosticScope scope = _client._clientDiagnostics.CreateScope($"{nameof(ContentDownloader)}.{nameof(DownloadStreamingInternal)}");
```

Reply:
I actually changed it to `ConversationClient.DownloadStreaming` since it is the origin of the call.

Reply:
I think you should have a scope in the method from the public class that matches the public method and for this internal class, you would use the name of the internal type. @christothes, @pakrym please correct me if I am mistaken.

Reply:
Correct, the diagnostic scopes are intended for customer distributed tracing scenarios and should represent client methods that the customer called not the implementation details.

Reply:
Ok, if I understood correctly what you are suggesting is that I should have both: In the public method I should have a scope and in the internal another different one. Both with their actual names and not referencing one another.
Is that correct?

Reply:
No, the outer(public method) scope is enough.

Reply:
Gotcha, thanks @pakrym.